### ISSUE TITLE
Javadoc generation fails via dokka-gradle-plugin (v0.9.9)

### ISSUE THREAD

**Issue by codeprogression** [Kotlin/dokka#90]
Given the closure ...

``` gradle
dokka {
    moduleName = 'data'
    outputFormat = 'javadoc'
    outputDirectory = "$buildDir/docs/ktdoc"
    linkMapping {
        dir = "src/main/kotlin"
        url = "https://internal/path/to/repo/src/main/kotlin"
    }
    sourceDirs = files('src/main/kotlin')
}
```

... Receiving the following output when running `./gradlew dokka`:

```
generation completed successfully
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
com.sun.tools.doclets.internal.toolkit.util.DocletAbortException: java.lang.StackOverflowError
```
- I have no `@see` tags defined
- Not all html files are created (causing broken links). e.g, `index.html` is never created
- When `outputFormat` is not specified, default html files are created correctly

_Let me know if you need any more information. Thanks!_

**Comment by yole** [comment]
I'd need to see the source file which causes the problem

**Comment by codeprogression** [comment]
@yole, after further investigation, these are actually two separate error conditions. One is fatal to the javadoc generation, the other isn't.

The stack overflow error is caused if I apply an annotation to an interface: 

``` kotlin
package demo

/**
 * This class supports greeting people by name.
 *
 * @property name The name of the person to be greeted.
 */
class GreeterImpl(val name: String): Greeter {

    /**
     * Prints the greeting to the standard output.
     */

    override fun greet() {
        println("Hello $name!")
    }

}

/**
 * Implementers will greet people.
 */
@Beta
interface Greeter {
    fun greet()
}

fun main(args: Array<String>) {
    demo.GreeterImpl(args[0]).greet()
}

@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CONSTRUCTOR, AnnotationTarget.FIELD, AnnotationTarget.FUNCTION,
        AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.CLASS, AnnotationTarget.FILE)
@MustBeDocumented
annotation class Beta
```

In this case, the documentation halts. Applying an annotation to the interface methods does not cause issues. (The stack overflow does not occur when the output format is "HTML," only "javadoc")

---

The  "null tag" error occurs if a referenced class has the `@suppress` comment added:

``` kotlin
package demo

/**
 * This class supports greeting people by name.
 * Inherits from [Greeter] which has a `suppress` doc comment
 *
 * @property name The name of the person to be greeted.
 */
class GreeterImpl(val name: String): Greeter {

    /**
     * Prints the greeting to the standard output.
     */

    override fun greet() {
        println("Hello $name!")
    }

}

/**
 * @suppress
 */
interface Greeter {
    fun greet()
}

fun main(args: Array<String>) {
    demo.GreeterImpl(args[0]).greet()
}
```

In this case, the documentation completes despite the `null:-1:-1: Tag @see` messages.

---

If I come up with additional cases, I will let you know. Hope that helps!

**Comment by yole** [comment]
Thanks for your investigation, it definitely does help! I'll look into these issues soon.

**Comment by codeprogression** [comment]
@yole  This is still happening in 0.9.10. Wondering if it will be worked on soon. Thanks.

**Comment by codeprogression** [comment]
@semoro @yole Thanks for working on this one!

**Comment by Xerus2000** [comment]
I still get spammed to death by `null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method` even though I have no @see tag ANYWHERE. And the error message isn't any help either...

**Comment by mstachniuk** [comment]
Same to me (using version 0.9.17). You can check my project: https://github.com/mstachniuk/graphql-schema-from-introspection-generator
`./gradlew clean build`

```
> Task :graphql-schema-from-introspection-generator-core:dokka
null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.
```

**Comment by semoro** [comment]
Problem with `@see` tags still reproduces, and should be fixed

**Comment by AlexanderLS** [comment]
Same to me (using version 0.9.17). You can check my project: https://github.com/Vorlonsoft/AndroidRate

**Comment by Xerus2000** [comment]
Guys, this is super annoying! My build log contains over 20 lines of this useless message:
`null:-1:-1: Tag @see cannot be used in inline documentation.  It can only be used in the following types of documentation: overview, package, class/interface, constructor, field, method.`
it isn't an actionable error, so please just remove it until is serves a purpose...

**Comment by JWT007** [comment]
Any word on this?  Since I tend to use @suppress on overridden hashCode/equals methods my build log if full of the null:-1:-1 errors ... is it just a simple mixup confusing @suppress with @see???

**Comment by JWT007** [comment]
So its not just the `@suppress`.

It is also caused by the documentation links `A [FooBar] implementation.` ... this gets resolved as a `@see`.

Removing the link references for a class/method eliminates the error.

**Comment by mdashlw** [comment]
Still relevant for me, I have no @see in my docs at all
Using maven plugin

**Comment by AmrAfifiy** [comment]
Seeing the same problem, happens when an interface B implements an interface A, with A having an annotation.

**Comment by pavelaizen** [comment]
Can we remove this logs from the console output?

**Comment by kamilok1965** [comment]
This will be fixed in 0.11.0 hopefully

**Comment by ccampo133** [comment]
Running into this as well. Happens when you reference classes/functions using brackets, e.g.

```
/**
 * See [foo] and [bar]
 */
```

**Comment by Kordyjan** [comment]
As far as I know, it is fixed in 1.4.0. If the problem is persisting it is ok to open a new issue as it will definitely have a different root cause that this problem.
